specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/blob.json

```diff
@@ -0,0 +1,1541 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Extends container ImmutabilityPolicy operation completed successfully..",
+            "schema": {
+              "$ref": "#/definitions/ImmutabilityPolicy"
+            },
+            "headers": {
+              "ETag": {
+                "description": "The ETag HTTP response header. This is an opaque string. You can use it to detect whether the resource has changed between requests. In particular, you can pass the ETag to one of the If-Match or If-None-Match headers.",
+                "type": "string"
+              }
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/blobServices/default/containers/{containerName}/lease": {
+      "post": {
+        "tags": [
+          "BlobContainers"
+        ],
+        "operationId": "BlobContainers_Lease",
+        "x-ms-examples": {
+          "Acquire a lease on a container": {
+            "$ref": "./examples/BlobContainersLease_Acquire.json"
+          },
+          "Break a lease on a container": {
+            "$ref": "./examples/BlobContainersLease_Break.json"
+          }
+        },
+        "description": "The Lease Container operation establishes and manages a lock on a container for delete operations. The lock duration can be 15 to 60 seconds, or can be infinite.",
+        "parameters": [
+          {
+            "$ref": "./storage.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "$ref": "#/parameters/ContainerName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "schema": {
+              "$ref": "#/definitions/LeaseContainerRequest"
+            },
+            "description": "Lease Container request body."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- Lease Container operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/LeaseContainerResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/blobServices/default/containers/{containerName}/migrate": {
+      "put": {
```

Comment:
>put [](http://example.com/codeflow?start=7&length=3)

This looks more like a POST action than a PUT to me. Any reason you chose PUT?

Reply:
This API follows the PUT semantics, it is completely idempotent and will update the state of the existing resource whenever required when invoked multiple times.